scripts/automated_ingestion/eessi_task.py

```diff
@@ -0,0 +1,1201 @@
+                                "PR '%s' is closed and not merged, returning REJECTED state", pull_request)
+                    # TODO: there is the possibility that the PR was updated just before the
+                    #         PR was closed
+                    #       WHY is it a problem? because a task may have been rejected that wouldn't
+                    #         have been rejected or worse shouldn't been rejected
+                    #       WHAT to do? ACCEPT/IGNORE THE ISSUE FOR NOw
+                    #       HOWEVER, the contents of the PR directory may be inconsistent with
+                    #         respect to the TaskState file and missing TaskSummary.html file
+                    #       WE could create an issue and only return EESSITaskState.REJECTED if the
+                    #         issue is closed
+                    #       WE could also defer all handling of this to the handler for the
+                    #         REJECTED state
+                    # FOR NOW, we assume that the task was rejected on purpose
+                    #   we need to change the state of the task in the default branch to REJECTED
+                    self._update_task_state_file(EESSITaskState.REJECTED)
+                    return EESSITaskState.REJECTED
+            else:
+                log_message(LoggingScope.TASK_OPS_DETAILS, "INFO",
+                            "PR '%s' is open, returning PULL_REQUEST state", pull_request)
+                return EESSITaskState.PULL_REQUEST
+        else:
+            log_message(LoggingScope.TASK_OPS_DETAILS, "INFO",
+                        "no PR found for branch '%s'", feature_branch_name)
+            # the method was called because the state of the task is PULL_REQUEST in the default branch
+            # however, it's weird that the PR was not found for the feature branch
+            # TODO: may create or update an issue for the task or deployment
+            return EESSITaskState.PULL_REQUEST
+
+        return EESSITaskState.PULL_REQUEST
+
+    @log_function_entry_exit()
+    def _perform_task_action(self) -> bool:
+        """Perform the task action"""
+        # TODO: support other actions than ADD
+        if self.action == EESSITaskAction.ADD:
+            return self._perform_task_add()
+        else:
+            raise ValueError(f"Task action '{self.action}' not supported (yet)")
+
+    @log_function_entry_exit()
+    def _issue_exists(self, title: str, state: str = "open") -> bool:
+        """
+        Check if an issue with the given title and state already exists.
+        """
+        issues = self.git_repo.get_issues(state=state)
+        for issue in issues:
+            if issue.title == title and issue.state == state:
+                return True
+        else:
+            return False
+
+    @log_function_entry_exit()
+    def _perform_task_add(self) -> bool:
+        """Perform the ADD task action"""
+        # TODO: verify checksum here or before?
```

Comment:
I think a checksum should be checked as early as possible, i.e. when it tries to download the payload. If the checksum isn't ok, the payload download failed, and one cannot reasonably move on to the next state. Reasonable thing to do would be to try and redownload x-times, then just forgot about it and try again the next time `ingest_bundles.py` is run.